### PR TITLE
search NFTs (scrum41) and users/TeaBags (scrum42)

### PR DESCRIPTION
**Pull request by ouafaema**
As a MINTER, I can search NFTs.

- using #hashtags
- using @username
- using any terms in the NFT description
- using a price range
- using location

As a MINTER, I can search users/TEA BAGS.

- using @username
- using location

Please do not merge this PR before the setting page PR to avoid conflicts.

### CODE REVIEW

**Review by fourm1 — APPROVED** [review]
Seems good

File: app/search/page.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+              >
+                <ProfileList data={profilesList}/>
+              </InfiniteScroll>
+            </TabsContent>
+          </div>
+        </Tabs>
+      </div>
+    </RightPanel>
+  )
+}
```
Comment (sckaper) [inline app/search/page.tsx:221]: there should be a message if there are no nfts/users found

File: src/repository/index.ts [hunk 2]
```diff
@@ -706,3 +710,46 @@ export async function registerUser(req: RegisterUserRequest) {
...
+
+export async function getPaginatedNftsWithSearch(query: string, location: string, priceRange: number[], page: number) {
+  const queryParams = new URLSearchParams({
+    query,
+    minPrice: priceRange[0].toString(),
+    maxPrice: priceRange[1].toString(),
+    location,
+    page: page.toString()
+  })
+  const response = await fetch(`/api/nft/search?${queryParams.toString()}`)
```
[the remark this answers is not in the document]
Comment (ouafaema) [reply src/repository/index.ts:722]: image

File: app/api/nft/search/route.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return problem({...invalidQueryParameterProblem, detail: "page must be a number"})
+  }
+
+  if (parsedPage <= 0) {
+    return problem({...invalidQueryParameterProblem, detail: "page must be a minimum 1"})
+  }
+
+  const session = getSession(req)
+
+  if (!session) {
```
[the remark this answers is not in the document]
Comment (ouafaema) [reply app/api/nft/search/route.ts:32]: yes it's critical

**Review by ouafaema — COMMENTED** [review]
changes done

**Review by ouafaema — COMMENTED** [review]
changes done

**Comment by Fenrur** [comment]
There are still some parts where I don't agree, but you can merge and refactor later

**Review by Fenrur — APPROVED** [review]
@ouafaema There are still some parts where I don't agree, but you can merge and refactor later

**Pull request merged**